catalog/spatial/wps/spatial-process-api/src/main/java/org/codice/ddf/spatial/process/api/description/AbstractDataDescription.java

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.spatial.process.api.description;
+
+import java.math.BigInteger;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import javax.ws.rs.core.MediaType;
+
+/** This class is Experimental and subject to change */
+public class AbstractDataDescription implements DataDescription {
+  private String id;
+
+  private String name;
+
+  private String description;
+
+  private BigInteger minOccurs = BigInteger.ONE;
+
+  private BigInteger maxOccurs = BigInteger.ONE;
+
+  private Metadata metadata;
+
+  private List<DataFormatDefinition> dataFormats =
+      Collections.singletonList(new DataFormatDefinition(MediaType.APPLICATION_OCTET_STREAM));
+
+  public AbstractDataDescription(String id, String name, String description) {
+    this.id = id;
+    this.name = name;
+    this.description = description;
+  }
+
+  public AbstractDataDescription(
+      String id, String name, String description, List<DataFormatDefinition> dataFormats) {
+    this(id, name, description);
+    this.dataFormats = new ArrayList<>(dataFormats);
+  }
+
+  @Override
+  public String getId() {
```

Comment:
going to remove the setters for this not needed I had them before I updated the constructor and it's now required-ish since it is in all of the constructors and isn't nullable